src/Utils/Rotation.cpp

```diff
@@ -455,89 +454,85 @@ void utils::Rotation::checkUnitary()
             utils::Matrix3d normalEq = Mm.transpose() * Mm;
             utils::Vector3d eigen_vals;
             utils::Matrix3d eigen_vecs;
-
-            /////////////////////
-            //eigen_sym(normalEq, eigen_vals, eigen_vecs);
-
             utils::Matrix3d V;
             utils::Vector3d d;
 
             int n = 3;
 
             utils::Vector3d e;
-            for (int i = 0; i < n; i++) {
-                for (int j = 0; j < n; j++) {
+            for (size_t i = 0; i < n; ++i) {
+                for (size_t j = 0; j < n; ++j) {
                     V(i, j) = normalEq(i, j);
                 }
             }
 
 
             // Symmetric Householder reduction to tridiagonal form. (tred2)
-            for (int j = 0; j < n; j++) {
+            for (size_t j = 0; j < n; ++j) {
                 d(j) = V(n-1, j);
             }
```

Comment:
`d = V.block(n-1, 0, 1, n);` ?

Reply:
I won't check anymore (I let you), but it feels that every matrix/vector operations were written by hand instead of accessing them directly, which is highly ineffective when using the Eigen library